addons/calendar/models/calendar_event.py

```diff
@@ -814,6 +814,48 @@ def copy(self, default=None):
             new_event.write({'partner_ids': [(Command.set(old_event.partner_ids.ids))]})
         return new_events
 
+    def action_unlink_event(self, attendee_id=None, recurrence=False):
+        """
+        Delete the event after displaying the delete wizard if necessary.
+
+        :param attendee_id: The ID of the attendee for the event
+        :param recurrence: Boolean indicating if the event is recurring
+
+        :return: Action to delete the event
+        """
+        if self.user_id._has_any_active_synchronization() or len(self.ids) > 1:
+            self.unlink()
+            return {
+                'type': 'ir.actions.act_url',
+                'target': 'self',
+                'url': '/odoo/calendar'
+            }
+
+        template = self.env.ref('calendar.calendar_template_delete_event', raise_if_not_found=False)
+        lang = self.env.context.get('lang')
+        if template.lang:
```

Comment:
Your code will crash when the template is not found. Please, can you test the changes you are making ? We need to be more resilient and have less fragile code.

```suggestion
        template = self.env.ref('calendar.calendar_template_delete_event', raise_if_not_found=False)
        lang = self.env.context.get('lang')
        if template and template.lang:
```
When the ref is not found, it will return None. 
![image](https://github.com/user-attachments/assets/3d3819b5-e17a-4ce1-a690-1d5228fba415)

Reply:
> I tried reproducing the error by changing the template name, but I couldn’t find any other case where it occurred. Could you suggest how to generate this error to make sure it’s handled properly in the future?
It is possible to delete mail.template records. Is this so difficult to imagine ? An administrator removing a template used in python code ?
![image](https://github.com/user-attachments/assets/f2094e1b-5814-4ad6-8625-98e443808d5e)